### PR TITLE
driver: wifi: Allow only one TWT session at any time.

### PR DESCRIPTION
**Pull request by ajayparida**
[SHEL-1392] Reject TWT setup command if,
already a TWT session is active.

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 1]
```diff
@@ -577,6 +581,11 @@ int wifi_nrf_set_twt(const struct device *dev,
 		status = wifi_nrf_fmac_twt_setup(rpu_ctx_zep->rpu_ctx,
 					   vif_ctx_zep->vif_idx,
 					   &twt_info);
+
+		if (status == WIFI_NRF_STATUS_SUCCESS) {
+			vif_ctx_zep->twt_in_progress = true;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:586]: what if the TWT setup is rejected? Will the application teardown explicitly?

Comment (ajayparida) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:586]: If TWT setup is rejected there will be no active session, 
application need not explicitly tear down the session.
'twt_in_progress' will be set to false, if for some reason twt setup failed/rejected.

**Review by rado17 — APPROVED** [review]

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c (hunk 1 above)
Comment (ajayparida) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:586]: I think, I mentioned ''need not" tear down the session explicitly

Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:586]: Ah sorry missed the *not*, and we are setting to false in the event, thanks.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 2]
```diff
@@ -553,6 +553,10 @@ int wifi_nrf_set_twt(const struct device *dev,
 
 	switch (twt_params->operation) {
 	case WIFI_TWT_SETUP:
+		if (vif_ctx_zep->twt_in_progress) {
+			return WIFI_TWT_FAIL_OPERATION_IN_PROGRESS;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:557]: Where is this defined? I guess you need a downstream commit for https://github.com/zephyrproject-rtos/zephyr/pull/55581/files

Comment (ajayparida) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:557]: Yes, I will do that before giving merge request for this one

**Review by sachinthegreen — APPROVED** [review]

**Comment by krish2718** [comment]
merged to #11079 to make Zephyr manifest udpate easier.

**Pull request closed without merging**